lib/engine/game/g_1880/step/parrer.rb

```diff
@@ -0,0 +1,138 @@
+            max_percent = @parring[:corporation] == @game.bcr ? 20 : max_shares * @parring[:corporation].share_percent
+            { 20 => '20%', 30 => '30%', 40 => '40%' }.select { |k, _v| k <= max_percent }
+          end
+
+          def process_choose(action)
+            case @parring[:state]
+            when :choose_percent
+              process_presidents_percent_choice(action)
+            when :choose_permit
+              process_building_permit_choice(action)
+            else
+              raise GameError, 'No choices to make at this time'
+            end
+          end
+
+          def process_presidents_percent_choice(action)
+            percent = action.choice
+            unless president_percent_choices.include?(percent)
+              error_msg = "Invalid percentage (#{percent}). " \
+                          "Choices are #{president_percent_choices.values.join(',')}."
+              raise GameError, error_msg
+            end
+
+            @log << "#{@parring[:entity].name} selects #{percent}% presidency share"
+            corporation = @parring[:corporation]
+            if percent != corporation.presidents_percent
+              num_to_remove = (percent - corporation.presidents_percent) / corporation.share_percent
+              shares_to_remove = corporation.shares.select { |s| s.buyable && !s.president }.pop(num_to_remove)
+              shares_to_remove.each { |s| corporation.shares.delete(s) }
+              corporation.presidents_share.percent = percent
+            end
+
+            @parring[:state] = :choose_permit
+            permit_choices = @game.building_permit_choices(@parring[:corporation])
+            return if permit_choices.size > 1
+
+            process_building_permit_choice(Action::Choose.new(@parring[:entity], permit_choices.first))
+          end
+
+          def process_building_permit_choice(action)
+            permit = action.choice
+            unless @game.building_permit_choices(@parring[:corporation]).include?(permit)
+              error_msg = "Invalid building permit (#{permit}). " \
+                          "Choices are #{@game.building_permit_choices(@parring[:corporation]).join(',')}."
+              raise GameError, error_msg
+            end
+
+            @log << "#{@parring[:entity].name} selects #{permit} building permit"
+            @parring[:corporation].building_permits = action.choice
+
+            @parring[:state] = :par_corporation
+            process_par(Action::Par.new(@parring[:entity], corporation: @parring[:corporation],
+                                                           share_price: @parring[:share_price]))
+          end
+
+          def process_par(action)
+            if @parring
+              @parring = nil
+              return super
+            end
+
+            raise GameError, 'BCR must par at 100' if action.corporation == @game.bcr && action.share_price.price != 100
+
```

Comment:
why include that instead of auto_action when BCR is parred? the par is always 100 and it's always the top spot

Reply:
Discussed in slack. BCR can choose any of the 100 par slots.